grpc-native/src/main/java/org/ballerinalang/net/grpc/builder/syntaxtree/SyntaxTreeGen.java

```diff
@@ -0,0 +1,302 @@
+            }
+            for (Method method : service.getServerStreamingFunctions()) {
+                client.addMember(getServerStreamingFunction(method).getFunctionDefinitionNode());
+                client.addMember(getServerStreamingContextFunction(method).getFunctionDefinitionNode());
+                serverStreamingClasses.add(getServerStreamClass(method));
+            }
+            for (Method method : service.getBidiStreamingFunctions()) {
+                client.addMember(getStreamingClientFunction(method, true).getFunctionDefinitionNode());
+                bidirectionalStreamingClasses.add(getStreamingClientClass(method));
+            }
+            moduleMembers = moduleMembers.add(client.getClassDefinitionNode());
+
+            for (Class streamingClient : clientStreamingClasses) {
+                moduleMembers = moduleMembers.add(streamingClient.getClassDefinitionNode());
+            }
+            for (Class streamingServer : serverStreamingClasses) {
+                moduleMembers = moduleMembers.add(streamingServer.getClassDefinitionNode());
+            }
+            for (Class streamingBidirectional : bidirectionalStreamingClasses) {
+                moduleMembers = moduleMembers.add(streamingBidirectional.getClassDefinitionNode());
+            }
+
+            for (java.util.Map.Entry<String, String> caller : service.getCallerMap().entrySet()) {
+                moduleMembers = moduleMembers.add(getCallerClass(caller.getKey(), caller.getValue())
+                        .getClassDefinitionNode());
+            }
+            for (java.util.Map.Entry<String, Boolean> valueType : service.getValueTypeMap().entrySet()) {
+                if (valueType.getValue()) {
+                    moduleMembers = moduleMembers.add(getValueTypeStream(valueType.getKey()).getTypeDefinitionNode());
+                }
+                moduleMembers = moduleMembers.add(getValueType(valueType.getKey()).getTypeDefinitionNode());
+            }
+        }
+
+        for (java.util.Map.Entry<String, Message> message : stubFile.getMessageMap().entrySet()) {
+            for (ModuleMemberDeclarationNode messageNode : getMessageNodes(message.getValue())) {
+                moduleMembers = moduleMembers.add(messageNode);
+            }
+        }
+
+        for (EnumMessage enumMessage : stubFile.getEnumList()) {
+            moduleMembers = moduleMembers.add(getEnum(enumMessage).getEnumDeclarationNode());
+        }
+
+        // ROOT_DESCRIPTOR
+        Constant rootDescriptor = new Constant("string", "ROOT_DESCRIPTOR", stubFile.getRootDescriptor(),
+                false);
+        moduleMembers = moduleMembers.add(rootDescriptor.getConstantDeclarationNode());
+
+        // getDescriptorMap function
+        Function getDescriptorMap = new Function("getDescriptorMap");
+        getDescriptorMap.addReturns(
+                getParameterizedTypeDescriptorNode("map", SYNTAX_TREE_VAR_STRING)
+        );
```

Comment:
Shall we add root descriptor only if it is not empty

Reply:
Fixed in 4208c62